src/multilevel-atom.cpp

```diff
@@ -98,11 +98,11 @@ static bool invert(realnum *S, int p) {
   } // singular
 
   int lwork = -1;
-  realnum work1;
+  realnum work1 = 0.0;
   DGETRI(&p, S, &p, ipiv, &work1, &lwork, &info);
   if (info != 0) abort("error %d in DGETRI workspace query", info);
   lwork = int(work1);
-  realnum *work = new realnum[lwork];
+  realnum *work = new realnum[lwork]();
```

Comment:
I don't understand these parens?

Reply:
Oh, I see, it initializes the array to zero.

Why is this needed?  `DGETRI` doesn't require the `work` array to be initialized.

Reply:
Yes. I agree that it will not cause a problem as it is the output variable. But the MSan will complain the non-initialization before using the variable. Usually we always initialize variables when they are defined. If I remember correctly, this is the same reason for the other PR submitted for NanoComp/mpb.

Reply:
Fine; I wish MSan was a bit more smart about this, but I agree that it doesn't hurt to silence the warning here.